src/pynwb/__init__.py

```diff
@@ -200,17 +200,18 @@ def validate(**kwargs):
 
 class NWBHDF5IO(_HDF5IO):
 
-    @docval({'name': 'path', 'type': (str, Path), 'doc': 'the path to the HDF5 file'},
+    @docval({'name': 'path', 'type': (str, Path), 'doc': 'the path to the HDF5 file', 'default': None},
             {'name': 'mode', 'type': str,
-             'doc': 'the mode to open the HDF5 file with, one of ("w", "r", "r+", "a", "w-", "x")'},
+             'doc': 'the mode to open the HDF5 file with, one of ("w", "r", "r+", "a", "w-", "x")',
+             'default': 'r'},
```

Comment:
Does this change the behavior of ``NWBHDF5IO`` when ``path`` and ``mode`` are used as positional arguments?  I think the answer is no, since all other arguments already had defaults and since this adds defaults for both ``path`` and ``mode``.  Overall the changes look fine to me, however, I just wanted to confirm as this question affects what version for release this change should go to.

Reply:
The only thing we need to watch out for here is if we have an arg without a default *after* an arg with a default. In that case the arg without a default will be an earlier positional arg. This actually came up in the partner PR. I had to move manager later so that it maintained the same position. It shouldn't be a problem here though.

Reply:
Sounds good